tree.go

```diff
@@ -1830,8 +1838,28 @@ func (n *LeafNode) serializeLeafWithUncompressedCommitments(cBytes, c1Bytes, c2B
 		copy(result[leafStemOffset:], n.stem[:StemSize])
 		copy(result[leafStemOffset+StemSize:], c1Bytes[:])
 		copy(result[leafStemOffset+StemSize+banderwagon.UncompressedSize:], cBytes[:])
-		copy(result[leafStemOffset+StemSize+2*banderwagon.UncompressedSize:], n.values[1])                                 // copy balance
-		copy(result[leafStemOffset+StemSize+2*banderwagon.UncompressedSize+leafBalanceSize:], n.values[2][:leafNonceSize]) // copy nonce
+		copy(result[leafStemOffset+StemSize+2*banderwagon.UncompressedSize:], n.values[0]) // copy basic data
+	case gapcount < 16:
+		// If there are less than 16 gaps, it's worth using skiplists
+		result = make([]byte, 1, nodeTypeSize+StemSize+bitlistSize+3*banderwagon.UncompressedSize+len(children))
+		result[0] = skipListType
+		result = append(result, n.stem[:StemSize]...)
+		result = append(result, cBytes[:]...)
+		result = append(result, c1Bytes[:]...)
+		result = append(result, c2Bytes[:]...)
+		var leafIdx int
+		for _, gap := range gaps {
+			if gap.Count == 0 {
+				break // skip the last gap as nothing follows
+			}
```

Comment:
Could we do:
```
for j:=0; j<=gapCount; j++ {
```
and avoid `if` L1852 ?

Reply:
not completely, because if `Count == 0` it means that the group ends with a gap, wherease if `Count != 0` then the group ends with a range. But yeah, no point in going over the whole list, I'll loop on `j`.